File: ATSPM/Data/Models/SpeedManagementConfigModels/Impact.cs
namespace ATSPM.Data.Models.SpeedManagementConfigModels
{
    public class Impact
    {
        public Guid? Id { get; set; }
        public string? Description { get; set; }
        public DateTime Start { get; set; }
        public DateTime? End { get; set; }
        public double StartMile { get; set; }
        public double EndMile { get; set; }
        public Guid ImpactTypeId { get; set; }
        public ImpactType? ImpactType { get; set; }
        public DateTime? CreatedOn { get; set; }
        public string? CreatedBy { get; set; }
        public DateTime? UpdatedOn { get; set; }
        public string? UpdatedBy { get; set; }
        public DateTime? DeletedOn { get; set; }
        public string? DeletedBy { get; set; }
        public List<Guid>? SegmentIds { get; set; }
    }
}

File: ATSPM/WebUI/src/features/speedManagementTool/types/impact.ts

export interface Impact {
    id?: string | null;
    description?: string | null;
    start?: Date | null;
    end?: Date | null;
    startMile: number;
    endMile: number;
    impactTypeIds?: string[] | null;
    impactTypes?: ImpactType[] | null;
    createdOn?: Date | null;
    createdBy?: string | null;
    updatedOn?: Date | null;
    updatedBy?: string | null;
    deletedOn?: Date | null;
    deletedBy?: string | null;
    segmentIds?: string[] | null;
}

export interface ImpactType {
    id?: string | null;
    name: string;
    description?: string | null;
}
